src/Handshake/ServerNegotiator.php

```diff
@@ -55,20 +55,38 @@ public function handshake(RequestInterface $request) {
             return new Response(400);
         }
 
+        $upgradeSuggestion = [
+            'Connection'             => 'Upgrade',
+            'Upgrade'                => 'websocket',
+            'Sec-WebSocket-Version'  => $this->getVersionNumber(),
+            'Sec-WebSocket-Protocol' => implode(', ', $this->_supportedSubProtocols)
+        ];
         if (true !== $this->verifier->verifyUpgradeRequest($request->getHeader('Upgrade'))) {
-            return new Response(400, [], '1.1', null, 'Upgrade header MUST be provided');
+            return new Response(
+                426,
+                $upgradeSuggestion,
+                null,
+                '1.1',
+                'Upgrade header MUST be provided'
+            );
         }
 
         if (true !== $this->verifier->verifyConnection($request->getHeader('Connection'))) {
-            return new Response(400, [], '1.1', null, 'Connection header MUST be provided');
+            return new Response(400, [], null, '1.1', 'Connection Upgrade MUST be requested');
         }
 
         if (true !== $this->verifier->verifyKey($request->getHeader('Sec-WebSocket-Key'))) {
-            return new Response(400, [], '1.1', null, 'Invalid Sec-WebSocket-Key');
+            return new Response(400, [], null, '1.1', 'Invalid Sec-WebSocket-Key');
         }
 
         if (true !== $this->verifier->verifyVersion($request->getHeader('Sec-WebSocket-Version'))) {
-            return new Response(426, ['Sec-WebSocket-Version' => $this->getVersionNumber()]);
+            /*
+             * https://tools.ietf.org/html/rfc7230#section-6.7
+             * A server that sends a 426 (Upgrade Required) response MUST send an
+             * Upgrade header field to indicate the acceptable protocols, in order
+             * of descending preference
+             */
+            return new Response(426, $upgradeSuggestion);
```

Comment:
Awesome. Fixes a violation of HTTP that wasn't explicitly documented in the RFC6455 spec.